server/src/main/java/org/opensearch/action/search/TransportDeletePitAction.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.GroupedActionListener;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Transport action for deleting pit reader context - supports deleting list and all pit contexts
+ */
+public class TransportDeletePitAction extends HandledTransportAction<DeletePitRequest, DeletePitResponse> {
+    private final NamedWriteableRegistry namedWriteableRegistry;
+    private TransportSearchAction transportSearchAction;
+    private final ClusterService clusterService;
+    private final SearchTransportService searchTransportService;
+    private static final Logger logger = LogManager.getLogger(TransportDeletePitAction.class);
+
+    @Inject
+    public TransportDeletePitAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedWriteableRegistry namedWriteableRegistry,
+        TransportSearchAction transportSearchAction,
+        ClusterService clusterService,
+        SearchTransportService searchTransportService
+    ) {
+        super(DeletePitAction.NAME, transportService, actionFilters, DeletePitRequest::new);
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportSearchAction = transportSearchAction;
+        this.clusterService = clusterService;
+        this.searchTransportService = searchTransportService;
+    }
+
+    /**
+     * Invoke 'delete all pits' or 'delete list of pits' workflow based on request
+     */
+    @Override
+    protected void doExecute(Task task, DeletePitRequest request, ActionListener<DeletePitResponse> listener) {
+        List<String> pitIds = request.getPitIds();
+        if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
```

Comment:
what happens if some one passes PIT Id1, PIT Id2, _all i.e. size!=1 but contains _all ?

Reply:
we will throw exception when '_all' is passed as part of body as its not a valid encoded pit id. Current implementation accepts '_all' as part of params as its in line with other similar impl such as clear scrolls.